Lib/typing.py

```diff
@@ -2042,8 +2042,17 @@ class Leaf:
       class Other(Leaf):  # Error reported by type checker
           ...
 
-    There is no runtime checking of these properties.
+    There is no runtime checking of these properties. The decorator
+    sets the ``__final__`` attribute to ``True`` on the decorated object
+    to allow runtime introspection.
     """
+    try:
+        f.__final__ = True
+    except (AttributeError, TypeError):
```

Comment:
I'd be open to just doing just `except Exception` here, in case some type throws a different exception if you try to set an attribute. For precedent, note that `@no_type_check` catches only `TypeError`.